doc/neps/nep-0021-advanced-indexing.rst

```diff
@@ -256,14 +359,30 @@ Alternatives
 ------------
 
 NumPy may not choose to offer these different type of indexing methods, or
-choose to only offer them through specific function instead of the proposed
+choose to only offer them through specific functions instead of the proposed
 notation above.
-For variations see also the open questions section above.
 
+We don't think that new functions are a good alternative, because indexing
+notation ``[]`` offer some syntactic advantages in Python (i.e., direct
+creation of slice objects) compared to functions.
```

Comment:
We already have hacks for this like `np.s_[some:slice]`. Is there a counter against `np.oindex[arr, the:actual:index]` or `arr[np.oindex[the:actual:index]]`? If nothing else, I'd perhaps like to see these listed as rejected alternatives.

Reply:
However, if we implement a special index object, which would not be unreasonable, I am not sure the point below means too much, since it would not be much of a function call, it would just bind to `arr` and then call `arr.__getitem__`, it would not actually do anything with `arr` itself. `arr` could be in charge of doing some of the checks (such as that the dimension is good) inside its `__getitem__`. Or maybe easier, `arr` has to provide `ndim` or maybe even `.shape` but that would be the only thing.